From 9388239954511d92d47fbe8cecb0b8cc76db124c Mon Sep 17 00:00:00 2001
From: Meng Wei <wei.meng@intel.com>
Date: Mon, 14 Jan 2019 10:29:04 +0800
Subject: [PATCH 059/292] media: intel-ipu4: [VIRT] Close dmabuf fd when dmabuf
 released.

For the case of virtualization, dmabuf is
created at service OS, therefore FD is only
known and can only be closed at service OS.

Signed-off-by: Ong Hock Yu <ong.hock.yu@intel.com>
Signed-off-by: Meng Wei <wei.meng@intel.com>
---
 drivers/media/pci/intel/ipu-psys.c | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/drivers/media/pci/intel/ipu-psys.c b/drivers/media/pci/intel/ipu-psys.c
index 5cc60bf97c57..f342951ee550 100644
--- a/drivers/media/pci/intel/ipu-psys.c
+++ b/drivers/media/pci/intel/ipu-psys.c
@@ -13,6 +13,9 @@
 #include <linux/mm.h>
 #include <linux/module.h>
 #include <linux/pm_runtime.h>
+#if defined(CONFIG_VIDEO_INTEL_IPU_ACRN) && defined(CONFIG_VIDEO_INTEL_IPU_VIRTIO_BE)
+#include <linux/syscalls.h>
+#endif
 #include <linux/version.h>
 #include <linux/poll.h>
 #if LINUX_VERSION_CODE < KERNEL_VERSION(4, 14, 0)
@@ -532,6 +535,9 @@ static int ipu_psys_release(struct inode *inode, struct file *file)
 				kbuf->dbuf = NULL;
 				kbuf->db_attach = NULL;
 				dma_buf_put(dbuf);
+#if defined(CONFIG_VIDEO_INTEL_IPU_ACRN) && defined(CONFIG_VIDEO_INTEL_IPU_VIRTIO_BE)
+				ksys_close(kbuf->fd);
+#endif
 			} else {
 				if (kbuf->db_attach)
 					ipu_psys_put_userpages(
-- 
2.21.0

